### PR TITLE
feat: render dm messages in a virtualized list

### PR DESCRIPTION
**Pull request by rygine**
## What does this PR do?

This PR is the first in a series that aim to optimize the performance of Lenster profile messages. It adds a virtualized list ([react-virtuoso](https://github.com/petyosi/react-virtuoso)) to render the conversation list. This prevents the expensive rendering of potentially thousands of DOM nodes at the same time.

During testing of this change, I discovered that long wallet addresses or profile names can cause some content to be clipped in the conversation list.

Screenshot 2023-04-25 at 2 32 01 PM

**Fixed version**

Screenshot 2023-04-25 at 1 55 29 PM

## Related issues

This somewhat relates to #837 as this change can be applied to those other lists.

This PR also relates to #2302, which will eventually be fixed by this series of PRs.

## Type of change

- [x] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [x] Enhancement (non-breaking small changes to existing functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] This change requires a documentation update

### <samp>🤖 Generated by Copilot at d6b92ca</samp>

*  Add `react-virtuoso` dependency to improve message preview list performance ([link](https://github.com/lensterxyz/lenster/pull/2700/files?diff=unified&w=0#diff-eda1f4100c8fae29a041adb14d95c174c2313e7056d3922bd0c0b1b77d634778R17), [link](https://github.com/lensterxyz/lenster/pull/2700/files?diff=unified&w=0#diff-eda1f4100c8fae29a041adb14d95c174c2313e7056d3922bd0c0b1b77d634778L144-R163), [link](https://github.com/lensterxyz/lenster/pull/2700/files?diff=unified&w=0#diff-32824c984905bb02bc7ffcef96a77addd1f1602cff71a11fbbfdd7f53ee026bbR291-R293), [link](https://github.com/lensterxyz/lenster/pull/2700/files?diff=unified&w=0#diff-32824c984905bb02bc7ffcef96a77addd1f1602cff71a11fbbfdd7f53ee026bbR16738-R16748))
*  Adjust CSS classes to prevent overflow and improve alignment of message preview component ([link](https://github.com/lensterxyz/lenster/pull/2700/files?diff=unified&w=0#diff-86a42fd57d0cd5557caf0ba432c3f36f048b7f3098460771c3f65de779dd14a4L39-R39), [link](https://github.com/lensterxyz/lenster/pull/2700/files?diff=unified&w=0#diff-86a42fd57d0cd5557caf0ba432c3f36f048b7f3098460771c3f65de779dd14a4L51-R58))
*  Update `pnpm-lock.yaml` file to record package metadata and ensure installation consistency ([link](https://github.com/lensterxyz/lenster/pull/2700/files?diff=unified&w=0#diff-32824c984905bb02bc7ffcef96a77addd1f1602cff71a11fbbfdd7f53ee026bbR291-R293), [link](https://github.com/lensterxyz/lenster/pull/2700/files?diff=unified&w=0#diff-32824c984905bb02bc7ffcef96a77addd1f1602cff71a11fbbfdd7f53ee026bbR16738-R16748))

## Emoji

🎨🚀➕

### CODE REVIEW

**Comment by bigint** [comment]
Is it possible to lazy load conversation in xmtp on scroll?

RN I think we are loading all at ones, consider a user having 10k convos, what will happen that time?

**Comment by bigint** [comment]
@rygine the alignment looks weird 🙇🏼 

image

File: apps/web/src/components/Messages/PreviewList.tsx [hunk 1]
```diff
@@ -141,22 +142,25 @@ const PreviewList: FC<PreviewListProps> = ({ className, selectedConversationKey
...
-                  isSelected={key === selectedConversationKey}
-                  key={key}
-                  profile={profile}
-                  conversationKey={key}
-                  message={message}
-                />
-              );
-            })
+            <Virtuoso
+              style={{ height: '100%' }}
```
Comment (bigint) [inline apps/web/src/components/Messages/PreviewList.tsx:146]: Do they support classNames? if so we can use tailwind class `h-full`

File: apps/web/src/components/Messages/PreviewList.tsx [hunk 2]
```diff
@@ -141,22 +142,25 @@ const PreviewList: FC<PreviewListProps> = ({ className, selectedConversationKey
...
+              data={sortedProfiles}
+              itemContent={(_, [key, profile]) => {
+                const message = messages.get(key);
+                if (!message) {
+                  return null;
+                }
+                return (
```
Comment (bigint) [inline apps/web/src/components/Messages/PreviewList.tsx:153]:
```suggestion
                  return null;
                }

                return (
```

**Review by bigint — CHANGES_REQUESTED** [review]

**Comment by rygine** [comment]
> the alignment looks weird 🙇🏼

@bigint fixed! [fix(apps/web): fix message preview styling](https://github.com/lensterxyz/lenster/pull/2700/commits/d08177e483843ca5e985f91fbdd8d84a718adfea)

File: apps/web/src/components/Messages/PreviewList.tsx (hunk 1 above)
Comment (rygine) [reply apps/web/src/components/Messages/PreviewList.tsx:146]: It looks like it does. Just tested locally with `h-full` and it seems to work just fine.

**Review by bigint — APPROVED** [review]
**LGTM** :shipit: 

Rad LGTM

**Comment by bigint** [comment]
Amazing thanks for the fix @rygine 🙏🏼

**Pull request merged**